src/main/java/com/laytonsmith/core/events/drivers/PlayerEvents.java

```diff
@@ -910,6 +910,67 @@ public boolean matches(Map<String, Mixed> prefilter, BindableEvent e) throws Pre
 			return false;
 		}
 
+		@Override
+		public PrefilterBuilder getPrefilters() {
+			return PrefilterBuilder
+					.start("button", PrefilterType.STRING_MATCH, new PrefilterMatcher() {
+						@Override
+						public String getStringParameter(BindableEvent e) {
+							if(e instanceof MCPlayerInteractEvent pie) {
+								if(pie.getAction().equals(MCAction.LEFT_CLICK_AIR)
+									|| pie.getAction().equals(MCAction.LEFT_CLICK_BLOCK)) {
+									return "left";
+								}
+								if(pie.getAction().equals(MCAction.RIGHT_CLICK_AIR)
+										|| pie.getAction().equals(MCAction.RIGHT_CLICK_BLOCK)) {
+									return "right";
+								}
+							}
+							return null;
+						}
+					})
+					.set("itemname", PrefilterType.STRING_MATCH, new PrefilterMatcher() {
+						@Override
+						public String getStringParameter(BindableEvent e) {
+							if(e instanceof MCPlayerInteractEvent pie) {
+								return pie.getItem().getType().getName();
+							}
+							return null;
+						}
+					})
+					.set("block", PrefilterType.STRING_MATCH, new PrefilterMatcher() {
+						@Override
+						public String getStringParameter(BindableEvent e) {
+							if(e instanceof MCPlayerInteractEvent pie) {
+								return pie.getClickedBlock().getType().getName();
```

Comment:
This can be null. For string matches we can return the String "null" in that case, unless you have a better idea.